File: C#/CreationalDesignPattern/CreationalDesignPattern/Refactoring/AbstractFactory/Factories/MacOSFactory.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace CreationalDesignPattern.Refactoring.AbstractFactory
{
    public class MacOSFactory : GUIFactory
    {
        public IButton CreateButton()
        {
            return new MacOSButton();
        }

        public Checkbox CreateCheckbox()
        {
            return new MacOSCheckbox();
        }
    }
}

File: Java/CreationalDesignPattern/src/main/java/com/ute/refactoring/abstractfactory/factories/MacOSFactory.java
package com.ute.refactoring.abstractfactory.factories;

import com.ute.refactoring.abstractfactory.buttons.Button;
import com.ute.refactoring.abstractfactory.buttons.MacOSButton;
import com.ute.refactoring.abstractfactory.checkboxes.Checkbox;
import com.ute.refactoring.abstractfactory.checkboxes.MacOSCheckbox;

public class MacOSFactory implements GUIFactory{
    @Override
    public Button createButton() {
        return new MacOSButton();
    }

    @Override
    public Checkbox createCheckbox() {
        return new MacOSCheckbox();
    }
}
